wgpu-core/src/command/transfer.rs

```diff
@@ -594,6 +597,26 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         if destination_offset % wgt::COPY_BUFFER_ALIGNMENT != 0 {
             return Err(TransferError::UnalignedBufferOffset(destination_offset).into());
         }
+        if !device
+            .downlevel
+            .flags
+            .contains(wgt::DownlevelFlags::UNRESTRICTED_INDEX_BUFFER)
+            && (src_buffer.usage.contains(BufferUsages::INDEX)
+                || dst_buffer.usage.contains(BufferUsages::INDEX))
+        {
+            let non_index_non_copy_buffer_usages = wgt::BufferUsages::VERTEX
+                | wgt::BufferUsages::UNIFORM
+                | wgt::BufferUsages::INDIRECT
+                | wgt::BufferUsages::STORAGE;
+            if src_buffer.usage.contains(non_index_non_copy_buffer_usages)
```

Comment:
Contains is the wrong operation here, we want intersects (as we want any, not all)